### PR TITLE
[Bzip2] Upgrade to v1.0.8 and build for experimental platforms

### PR DESCRIPTION
**Pull request by giordano**
Before merging I want to do some tests to make sure I'm not going to break anything.

### CODE REVIEW

File: B/Bzip2/build_tarballs.jl [hunk 1]
```diff
@@ -25,35 +25,34 @@ make CFLAGS="${CFLAGS}" PREFIX=${prefix} install
...
-    $CC -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.6 $LDFLAGS $OBJS
-    ln -s libbz2.so.1.0.6 libbz2.so.1.0
-    ln -s libbz2.so.1.0.6 libbz2.so.1
-    ln -s libbz2.so.1.0.6 libbz2.so
-    mv libbz2.so* ${prefix}/lib/
+    cc -shared -Wl,-soname -Wl,libbz2.so.1.0 -o "libbz2.so.${VERSION}" ${OBJS}
+    ln -s "libbz2.so.${VERSION}" libbz2.so.1.0
+    ln -s "libbz2.so.${VERSION}" libbz2.so.1
+    ln -s "libbz2.so.${VERSION}" libbz2.so
 fi
```
Comment (giordano) [inline B/Bzip2/build_tarballs.jl:40]: For future reference, the makefile `Makefile-libbz2_so` might be able to replace most of this (but some symlinks are probably still missing)

**Comment by DilumAluthge** [comment]
> We need to cap the compatibility of bzip2 of all dependencies 😑

Instead of having to do the cap, could you instead make a breaking release of the JLL package? E.g. bump to 2.0.0 or something?

**Comment by DilumAluthge** [comment]
> > We need to cap the compatibility of bzip2 of all dependencies 😑
> 
> Instead of having to do the cap, could you instead make a breaking release of the JLL package? E.g. bump to 2.0.0 or something?

Ah, I just realized that my idea won't work, because most people don't have upper-bounded `[compat]` entries for JLL dependencies.

See also: https://github.com/JuliaRegistries/RegistryCI.jl/issues/394

**Comment by giordano** [comment]
> most people don't have upper-bounded [compat] entries for JLL dependencies.

Nor JLL packages, which are the main issue (there is a single non-JLL package depending on this, out of 606 dependents)

**Pull request merged**